src/Compilers/Core/Portable/SourceGeneration/ISyntaxReceiver.cs

```diff
@@ -36,5 +50,35 @@ public interface ISyntaxReceiver
     /// Allows a generator to provide instances of an <see cref="ISyntaxReceiver"/>
     /// </summary>
     /// <returns>An instance of an <see cref="ISyntaxReceiver"/></returns>
-    public delegate ISyntaxReceiver SyntaxReceiverCreator();
+    public delegate ISyntaxReceiverBase SyntaxReceiverCreator();
```

Comment:
Is this a breaking change?

Reply:
Technically it's a breaking public API change. Anything that uses a `SyntaxReceiverCreator` instance directly to create a receiver would potentially break, as the return type has changed.

In reality we would expect that only the generator driver internals are doing that. If the calling code is just passing an instance of the creator it's not a breaking change to change the return type. It can still return a more derived type and be a valid delegate.

IMO, its fine. Obviously if compat disagrees, we can add a new overload with a new delegate to maintain strict back compat.

Reply:
I would prefer we went with a new `delegate` here to avoid the potential for corner case compat breaks.